server/src/domain/server-info/server-info.service.ts

```diff
@@ -83,15 +86,23 @@ export class ServerInfoService {
     const loginPageMessage = process.env.PUBLIC_LOGIN_PAGE_MESSAGE || '';
 
     const isInitialized = await this.userRepository.hasAdmin();
+    const onboarding = await this.systemMetadataRepository.get(SystemMetadataKey.ADMIN_ONBOARDING);
 
     return {
       loginPageMessage,
       trashDays: config.trash.days,
       oauthButtonText: config.oauth.buttonText,
       isInitialized,
+      isOnboarded: onboarding?.isOnboarded || false,
     };
   }
 
+  async setAdminOnboarding(): Promise<boolean> {
+    await this.systemMetadataRepository.set(SystemMetadataKey.ADMIN_ONBOARDING, { isOnboarded: true });
+    const onboarding = await this.systemMetadataRepository.get(SystemMetadataKey.ADMIN_ONBOARDING);
+    return !!onboarding?.isOnboarded;
```

Comment:
This is probably not needed, right? Why does this need to return a boolean in the first place?

Reply:
Are you recommending we shouldn't check for the return value? just by status code?

Reply:
I think if the database write was successful, you can assume the value was saved and is true. I don't see why you need to re-query it again.